packages/flutter/lib/src/widgets/page_view.dart

```diff
@@ -191,13 +197,16 @@ class PageController extends ScrollController {
     int page, {
     required Duration duration,
     required Curve curve,
+    bool callPageChangeAtEnd = false,
```

Comment:
This parameter needs to be documented in the docs above.

Reply:
Having this parameter is strange API design. We should decide whether it should announce all pages along the way or not and then implement that.

Reply:
I've moved the documentation from the above variable to this constructor.

> Having this parameter is strange API design. We should decide whether it should announce all pages along the > way or not and then implement that.


I am making this choice here, this calls `onPageChanged` explicitly. 

```
          if (currentPage != _lastReportedPage) {
            if (!widget.controller._callPageChangeAtEnd  && notification is ScrollUpdateNotification) {
              _lastReportedPage = currentPage;
              widget.onPageChanged!(currentPage);
            } else if (widget.controller._callPageChangeAtEnd  && notification is ScrollEndNotification) {
              _lastReportedPage = currentPage;
              widget.onPageChanged!(currentPage);
            }
          }
```

Reply:
If a user wanted this, could they just listen for a ScrollEndNotification themselves?

Reply:
Thanks so much for the feedback, since the issue this change, i thought about out of the box solution 
Based on your feedback, closed issue with an update example https://github.com/flutter/flutter/issues/43813#issuecomment-990747010